nemo_curator/cache.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nemo_curator.utils.file_utils import expand_outdir_and_mkdir
+
+# Global variable to store the cache directory
+_global_cache_dir = None
```

Comment:
Instead of a global variable, I think we should the singleton design pattern here, where there is a `Cache` class that creates the static singleton instance (if it doesn't exist), or returns an existing one (if it was already created).

Subsequently, the init and get methods can be nicely wrapped inside that class.

Reply:
@sarahyurick please let me know what you think and whether my comment makes sense to you

Reply:
Thank you! Yes, that makes sense to me. Sorry I haven't gotten back to this earlier, but I expect to have more changes ready next week.